### ISSUE TITLE
Electron desktop "initializing R" flashes on each retry

### ISSUE THREAD

**Issue by gtritchie** [rstudio/rstudio#9627]
### System details

    RStudio Edition : Desktop
    RStudio Version : Electron (must currently build manually)
    OS Version      : macOS (probably all)
    R Version       : any

### Steps to reproduce the problem
Delay the start of rsession either by setting the environment variable RSTUDIO_SESSION_SLEEP_ON_STARTUP to a number, in seconds, to stall. For example: `RSTUDIO_SESSION_SLEEP_ON_STARTUP=5`.

Or, start RStudio with command-line switch: `--session-delay-seconds=5`.

When you start, after a brief pause you'll see the screen reading "Initializing R" below:

![screenshot of initializing R display](https://user-images.githubusercontent.com/10569626/126543409-84f9d72d-2708-4fa6-95bd-aa0383e9bfe0.png)

### Describe the problem in detail
This screen will flash blank each time RStudio retries the session load.

### Describe the behavior you expected
Should not do this. We either need to find a way to defer the screen update until the real page starts to load, or perhaps show this message in a separate modal window?

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by gtritchie** [comment]
More background for when I or somebody else gets to looking at this again:

We show this page (loaded from a file) when there’s a delay in the rsession responding during startup. We do a fixed number of timer-based retries (at increasing intervals) which load this status page, then fire off a reload of the real page URL. If it fails to respond again, we reload the status page. In Qt, this keeps showing the status page until the “real” page responds, but in Electron it immediately replaces the status page with a blank page, then shows the status page again. So… blinking.

I may very well be missing some obvious flag or approach here that would prevent the flashing. What I did try was using an Electron `BrowserView` overlaid over (same size and location) the regular view that's loading the "real" page. That did prevent the flashing, but `BrowserView` is marked as experimental, and I hit other issues that made me think it was too complex for solving this single-case we have of this.

**Comment by kevinushey** [comment]
I wonder if we can rework the flow here with something like:

1. Once we're ready to launch, check whether the `rsession` is available. If it is, load it directly; otherwise, load the status page.
2. If we loaded the status page, allow the status page to ping RStudio (via some Javascript?) until it reports the rsession is ready. Once it's ready, then we load the main page.

In other words, rather than relying on RStudio / Electron to refresh the status page periodically, we just load it once and let the status page take care of loading the main page. Does that make sense?

**Comment by gtritchie** [comment]
Yes, could look at something like that. Haven't thought it through, but might require some additional ipc callbacks using the context-bridge mechanism (the renderer hosting the status page doesn't have direct access to anything in the main process, by design). Was trying to avoid needing to modify the existing status page and leave it dumb, and replicate the existing behavior as close as possible.

Another way would be to popup the status page (or just a 'loading.... message') as a modal dialog over the main window. That would be more desktoppy.

Also have to look at how this behaves with desktop pro loading remote sessions, haven't dug into that yet.

I'd say this is fairly low priority, relative to everything else. We will probably be a lot smarter in a month or so and the answer will be blazingly obvious!

**Comment by gtritchie** [comment]
Looked at using a simple Dialog.MessageBox to display the "still loading" message over the window. ALMOST doable but can't currently programmatically dismiss a dialog (argh). That's coming in the next major version of Electron (15), probably too late for us.

https://github.com/electron/electron/pull/26102

**Comment by gtritchie** [comment]
This was fixed 6+ months ago, and if it regresses will be very noticable. Closing.
